test/prefer-both-either.js

```diff
@@ -0,0 +1,53 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/prefer-both-either';
+
+const ruleTester = avaRuleTester(test, {
+    env: {
+        es6: true
+    },
+    parserOptions: {
+        sourceType: 'module'
+    }
+});
+
+const prefer = {
+    allPass: 'both',
+    anyPass: 'either'
+}
+
+const error = (instead, args) => ({
+    ruleId: 'prefer-both-either',
+    message: `Instead of \`${instead}([${args}])\`, prefer \`${prefer[instead]}(${args})\``
+});
+
+ruleTester.run('prefer-both-either', rule, {
+    valid: [
+        'both(foo, bar)',
+        'either(foo, bar)',
+        'allPass([foo, bar, baz])',
+        'anyPass([foo, bar, baz])',
+        'allPass(predicates)',
+        'allPass(predicates, foo)',
+        'anyPass(predicates)',
+        'anyPass(predicates, foo)'
+    ],
+    invalid: [
+        {
+            code: 'allPass([foo, bar])',
+            errors: [error('allPass', 'foo, bar')]
```

Comment:
Can you add a test cases where the elements of the array are not identifiers?

For example: 

* `allPass([ (foo) => !foo, function () { return false; } ])` (arrow functions, anonymous function expression)
* `allPass([ R.T, R.F ])` (member expression)

I suspect the first case to fail due how the error reporting works.

Reply:
The elements could also be call expressions, e.g. `allPass([ complement(foo), complement(bar) ])`.

Reply:
I get your point, I will simplify the error reporting. I think the same problem exists in `prefer-complement`, will PR for this also.